server.py

```diff
@@ -1,23 +1,76 @@
 import socket
+import email.utils
 
-ADDR = ('127.0.0.1', 7000)
 
-socket = socket.socket(
-    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
-    )
+def response_ok(uri):
+    returnstr = []
+    content_type = "text/plain"
+    bytes = uri.encode('utf-8')
+    content_length = str(len(bytes))
+    date = email.utils.formatdate(usegmt=True)
+    returnstr.append("HTTP/1.1 200 OK")
+    returnstr.append("Date:" + date)
+    returnstr.append("Content-Type:" + content_type)
+    returnstr.append("Content-Length:" + content_length)
+    header = '\r\n'.join(returnstr)
+    header = header + "\r\n" + uri
+    return header
 
-socket.bind(ADDR)
-socket.listen(1)
 
-while True:
+def response_error():
+    return "500 Server Error \r\n\r\n"
+
+
+def parse_request(fullmsg):
     try:
-        conn, addr = socket.accept()
-        while True:
-            msg = conn.recv(16)
-            print msg
-            conn.sendall(msg)
-            if len(msg) < 16:
-                conn.close()
-                break
-    except KeyboardInterrupt:
-        break
\ No newline at end of file
+        tline1 = fullmsg[0]
+        line1 = tline1.split()
+        tline2 = fullmsg[1]
+        line2 = tline2.split()
```

Comment:
Here you are assuming that the `Host` header is going to be the first header after the initial line.  That is *not* a safe assumption.